java/src/main/java/ai/rapids/cudf/HostColumnVector.java

```diff
@@ -1283,6 +1435,40 @@ public Builder appendArray(double... values) {
       return this;
     }
 
+    public Builder appendUnscaledDecimalArray(int... values) {
+      assert type.typeId == DType.DTypeEnum.DECIMAL32;
```

Comment:
I find it surprisingly inconsistent that appending a single value will implicitly upcast to a long if the type is DECIMAL64 yet appending more than one value at a time does not.

Reply:
Because function family `appendUnscaledDecimalArray` is performance-oriented,  there seems no other way upcasting int array to long one but iterating the array. It is quite time-consuming comparing to memory copy.

Reply:
If we're so concerned about performance rather than semantics then why provide `appendUnscaledDecimal(int value)` that works for `DECIMAL64`?  Make the caller cast it to a long if they want a `DECIMAL64`.  That's what this code is doing, and I don't see any reason for these two methods of appending values to be inconsistent.  Eliminating the DECIMAL64 implicit support for appending single ints would technically make that method faster, if we're all about performance.  :wink: